Subject: [PATCH] Also remove activity record if its app is removed.

When forcing stop the package of top activity during screen off,
because the top activity is paused with state, its activity will
not be removed then continue to restart the activity once and
finish it again that is unnecessary.

diff --git a/services/core/java/com/android/server/am/ActivityStack.java b/services/core/java/com/android/server/am/ActivityStack.java
index a47e7c6..bbb5fce 100644
--- a/services/core/java/com/android/server/am/ActivityStack.java
+++ b/services/core/java/com/android/server/am/ActivityStack.java
@@ -4254,7 +4254,7 @@ boolean removeHistoryRecordsForAppLocked(ProcessRecord app) {
                         hasVisibleActivities = true;
                     }
                     final boolean remove;
-                    if ((!r.haveState && !r.stateNotNeeded) || r.finishing) {
+                    if ((!r.haveState && !r.stateNotNeeded) || r.finishing || app.removed) {
                         // Don't currently have state for the activity, or
                         // it is finishing -- always remove it.
                         remove = true;
